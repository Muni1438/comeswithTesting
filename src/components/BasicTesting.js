import React from 'react'

function BasicTesting() {
    return (
        <div>
            <h1 data-testId="h1Content">Hello there I'm back, and i'm came here with testing knowlwdge</h1>
        </div>
    )
}

export default BasicTesting
