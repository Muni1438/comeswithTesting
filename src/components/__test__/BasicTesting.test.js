import React from "react"
import {render} from "@testing-library/react"
import BasicTesting from "../BasicTesting"

describe("all about h1 tag",()=>{
    test('should render correctly', () => {
        const {asFragment} = render(<BasicTesting/>)
        expect(asFragment(<BasicTesting/>)).toMatchSnapshot()
    })
    test('should Have text content', () => {
        const {getByTestId} = render(<BasicTesting/>)
        expect(getByTestId("h1Content")).toHaveTextContent("Hello there I'm back, and i'm came here with testing knowlwdge")
        
    })
    
    
})
