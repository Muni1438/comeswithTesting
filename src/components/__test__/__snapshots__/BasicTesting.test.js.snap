// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`all about h1 tag should render correctly 1`] = `
<DocumentFragment>
  <div>
    <h1
      data-testid="h1Content"
    >
      Hello there I'm back, and i'm came here with testing knowlwdge
    </h1>
  </div>
</DocumentFragment>
`;

exports[`all about should render correctly 1`] = `
<DocumentFragment>
  <div>
    <h1
      data-testid="h1Content"
    >
      Hello there I'm back, and i'm came here with testing knowlwdge
    </h1>
  </div>
</DocumentFragment>
`;

exports[`should render correctly 1`] = `
<DocumentFragment>
  <div>
    <h1
      data-testid="h1Content"
    >
      Hello there I'm back, and i'm came here with testing knowlwdge
    </h1>
  </div>
</DocumentFragment>
`;
